sphinx.go

```diff
@@ -206,6 +207,20 @@ func NewOnionPacket(paymentPath *PaymentPath, sessionKey *btcec.PrivateKey,
 		hopPayloadBuf bytes.Buffer
 	)
 
+	// Read out random bytes to fill out the rest of the starting packet
+	// after the hop payload for the final node. This mitigates a privacy
+	// leak that may reveal a lower bound on the true path length to the
+	// receiver.
+	//
+	// TODO(roasbeef): generate a rand stream using the session key
+	// instead?
+	firstHopSize := paymentPath[numHops-1].HopPayload.NumBytes()
+	randBytes := make([]byte, routingInfoSize-firstHopSize)
+	if _, err := rand.Read(randBytes[:]); err != nil {
```

Comment:
read directly into `mixHeader[firstHopSize:]`, spares a large allocation and additional copying